pkg/render/intrusion_detection.go

```diff
@@ -892,3 +950,243 @@ func (c *intrusionDetectionComponent) intrusionDetectionAnnotations() map[string
 	}
 	return nil
 }
+
+// intrusionDetectionComponentResources sets the default requirements value for DPI in IDS resource if it is not set.
+func (c *intrusionDetectionComponent) intrusionDetectionComponentResources() {
```

Comment:
[Cosmetic] This function should contain the word `set` or `default` to signal that is cause side effects.